File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    people.sort()
    answer = 0
    p = deque(people)
    
    while p:
        if len(p)>1:
            if p[0] + p[-1] <= limit:
                p.popleft()
                p.pop()
                answer += 1
            else:
                p.pop()
                answer += 1
        else:
            answer += 1        
            break
    
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public static int solution(int[] people, int limit) {
        int answer = 0;

        ArrayDeque<Integer> arr = new ArrayDeque<>();
        Arrays.sort(people);
        for (int p : people) {
            arr.add(p);
        }
        
        while (!arr.isEmpty()) {
            if (arr.size() > 1) {
                if (arr.getFirst() + arr.getLast() <= limit) {
                    arr.removeFirst();
                    arr.removeLast();
                    answer++;
                    continue;
                }
                arr.removeLast();
                answer++;
            } else {
                answer ++;
                break;
            }
        }
        return answer;
    }
}